### PR TITLE
Add php version 8.0.16

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1522369
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: platforms/php/versionsToBuild.txt [hunk 1]
```diff
@@ -19,6 +19,7 @@
 8.0.3, c9816aa9745a9695672951eaff3a35ca5eddcb9cacf87a4f04b9fb1169010251, 1729F83938DA44E27BA0F4D3DBDB397470D12172 BFDDD28642824F8118EF77909B67A5C12229118F,
 8.0.11, e3e5f764ae57b31eb65244a45512f0b22d7bef05f2052b23989c053901552e16, 1729F83938DA44E27BA0F4D3DBDB397470D12172 BFDDD28642824F8118EF77909B67A5C12229118F,
 8.0.13, cd976805ec2e9198417651027dfe16854ba2c2c388151ab9d4d268513d52ed52, 1729F83938DA44E27BA0F4D3DBDB397470D12172 BFDDD28642824F8118EF77909B67A5C12229118F,
+8.0.16, f27a2f25259e8c51e42dfd74e24a546ee521438ad7d9f6c6e794aa91f38bab0a, 1729F83938DA44E27BA0F4D3DBDB397470D12172, BFDDD28642824F8118EF77909B67A5C12229118F,
```
Comment (qianz2) [inline platforms/php/versionsToBuild.txt:22]: Can we add dynamic installation test for php new version?

Comment (qianz2) [inline platforms/php/versionsToBuild.txt:22]: Can we add dynamic installation test for php new version?

**Review by qianz2 — APPROVED** [review]

File: src/BuildScriptGenerator/PhpVersions.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@ public static class PhpVersions
         public const string Php81TarSha256 = "a1317eff0723a2b3d3122bbfe107a1158570ea2822dc35a5fb360086db0f6bbc";
         public const string Php80Version = "8.0.17";
         public const string Php80Keys = "1729F83938DA44E27BA0F4D3DBDB397470D12172 BFDDD28642824F8118EF77909B67A5C12229118F";
-        public const string Php80TarSha256 = "bdbd792901c156c4d1710c9d266732d3c17f6ff63850d6660b9d8d3411188424";
+        public const string Php80TarSha256 = "f27a2f25259e8c51e42dfd74e24a546ee521438ad7d9f6c6e794aa91f38bab0a";
```
Comment (daniv-msft) [inline src/BuildScriptGenerator/PhpVersions.cs:20]: I'm a bit confused about the whole 8.0.17 vs. 8.0.16 thing. If we're already supporting 8.0.17, could you please clarify why we should support 8.0.16?
Also, could you please clarify why we're updating the SHA here if we're keeping the current PHP version as 8.0.17?

File: platforms/php/versionsToBuild.txt (hunk 1 above)
Comment (william-msft) [reply platforms/php/versionsToBuild.txt:22]: Updated. Thanks!

Comment (william-msft) [reply platforms/php/versionsToBuild.txt:22]: Updated. Thanks!

File: src/BuildScriptGenerator/PhpVersions.cs (hunk 2 above)
Comment (william-msft) [reply src/BuildScriptGenerator/PhpVersions.cs:20]: I confirmed with the team `8.0.17` is fine. Closing this PR

**Pull request closed without merging**